public/TreeHelpers.js

```diff
@@ -0,0 +1,226 @@
+$.fn.extend({
```

Comment:
Please use one anguage across the project - either coffee or JS. (I vote for coffee)